src/ovs/aca_ovs_control.cpp

```diff
@@ -0,0 +1,326 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_ovs_control.h"
+#include "ovs_control.h"
+#include <iostream>
+#include <vector> 
+#include <unordered_map> 
+#include <signal.h>
+#include <syslog.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <chrono>
+#include <errno.h>
+#include <stdexcept>
+#include <inttypes.h>
+// #include <unistd.h>
+#include <pcap.h>
+#include <net/ethernet.h>
+#include <netinet/ether.h>
+#include <netinet/ip.h>
+#include <arpa/inet.h>
+
+using namespace std;
+using namespace ovs_control;
+
+extern string g_ofctl_command;
+extern string g_ofctl_target;
+extern string g_ofctl_options;
+
+namespace aca_ovs_control
+{
+ACA_OVS_Control &ACA_OVS_Control::get_instance()
+{
+  // Instance is destroyed when program exits.
+  // It is instantiated on first use.
+  static ACA_OVS_Control instance;  
+
+  return instance;
+}
+
+int ACA_OVS_Control::control()
+{
+    int overall_rc = EXIT_SUCCESS;
+
+    char target[g_ofctl_target.size() + 1];
+    g_ofctl_target.copy(target, g_ofctl_target.size() + 1);
+    target[g_ofctl_target.size()] = '\0';
+
+    char options[g_ofctl_options.size() + 1];
+    g_ofctl_options.copy(options, g_ofctl_options.size() + 1);
+    options[g_ofctl_options.size()] = '\0';
+
+    if (g_ofctl_command.compare("monitor") == 0) {
+        monitor(target, options);
+    } else if (g_ofctl_command.compare("dump-flows") == 0) {
+        dump_flows(target, options);
+    } else if (g_ofctl_command.compare("packet-out") == 0) {
+        packet_out(target, options);
+    } else {
+        cout << "Usage: -c <command> -t <target> -o <options>" << endl;
+        cout << "   commands: monitor, dump-flows, packet-out..." << endl;
+        cout << "   target: swtich name, such as br-int, br-tun, ..." << endl;
+        cout << "   options: " << endl;
+        cout << "      moinor: \"[miss-len] [invalid-ttl] [resume] [watch:format]\"" << endl;
+        cout << "      packet-out: \"in_port=<in_port> packet=<hex string> [actions=<actions>]\"" << endl;
+    }
+
+    return overall_rc;
+}
+
+int ACA_OVS_Control::dump_flows(const char *bridge, const char *opt) 
+{
+    int overall_rc = EXIT_SUCCESS;
+    OVS_Control::get_instance().dump_flows(bridge, opt);
+    return overall_rc;
```

Comment:
OVS_Control::get_instance().dump_flows(bridge, opt) -> return void. I think the ACA_OVS_Control::dump_flows should return void also since there is no need to return EXIT_SUCCESS always. Same thing for the functions below.

Reply:
return type has been changed from int to void for dump_flows, monitor, and packet_out functions in aca_ovs_control.cpp